src/group_theory/cyclic.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2019 Neil Strickland. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Neil Strickland
+
+We define finite cyclic groups, in multiplicative notation.
+The elements of `Cₙ` are denoted by `r i` for `i : zmod n`.
+We prove that an element `g ∈ G` with `gⁿ = 1` gives rise to
+a homomorphism `Cₙ → G`.  We also do the case n = ∞ separately.
+-/
+
+import data.fintype algebra.power_mod
+
+namespace group_theory
+
+variable n : ℕ+
+
+@[derive decidable_eq]
+inductive cyclic
+| r : (zmod n) → cyclic
```

Comment:
Couldn't this just be `cyclic n := multiplicative (zmod n)`?

Reply:
I think that that would be notationally confusing in practice, and would also make the interaction with the dihedral group less pleasant.

Reply:
Why do you think that? Mario's suggestion will turn `cyclic n` into a group with multiplicative notation. Setting up the group instance would be only 1 line.